tests/unit/samplers/test_config.py

```
@@ -0,0 +1,277 @@
+    cfg = BlackJAXSMCConfig()
+    assert cfg.persistent_sampling is True
+    assert cfg.temperature_ladder is None
+
+
+def test_smc_temperature_ladder_valid():
+    cfg = BlackJAXSMCConfig(temperature_ladder=[0.0, 0.5, 1.0])
+    assert cfg.temperature_ladder == [0.0, 0.5, 1.0]
+
+
+def test_smc_temperature_ladder_must_start_at_zero():
+    with pytest.raises(ValidationError, match="start at 0.0"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)